harvester: decouple plot refresh from API requests

ctrlaltdel opened the pull request:
Moved plot refresh in its own asynchronous task, this helps prevent slow
responses on big harvesters (5K+ plots) when the plot refresh is triggered.

Fixes #5208.

mariano54 commented on chia/harvester/harvester_api.py:
```diff
@@ -82,9 +77,7 @@ def blocking_lookup(filename: Path, plot_info: PlotInfo) -> List[Tuple[bytes32,
             try:
                 plot_id = plot_info.prover.get_id()
                 sp_challenge_hash = ProofOfSpace.calculate_pos_challenge(
-                    plot_id,
```
Can you revert these automatic style changes to keep the diff small?

mariano54 commented on chia/harvester/harvester.py:
```diff
@@ -117,6 +120,15 @@ async def refresh_plots(self):
+        while True:
+
+            # Don't run plot refresh until farmer handshake was done
+            if self.farmer_public_keys and self.pool_public_keys:
```
If we start the refresh before handshake is done, we might wait a long time to load the plots. We should load the plots immediately after the handshake. Can we maybe have another loop before this while True?

wjblanke replied:
Hmm still some odd behavior going on with the CI tests

mariano54 commented:
Thanks for the change! made some comments

ctrlaltdel committed:
harvester: decouple plots refresh from API requests

Moved plots refresh in its own asynchronous task, this helps prevent slow
responses on big harvesters (5K+ plots) when the plot refresh is triggered.

wjblanke reviewed (dismissed):
aok

HarukaMa commented:
Looks like the farming is still being blocked - the harvester would just wait for scanning to finish before starting to process SPs. Those SPs received during the scan will still be delayed.

Log looks like this:

```
2021-06-18T03:58:47.318 harvester chia.harvester.harvester: INFO     ** plots were eligible for farming 4cac0ca524... Found 0 proofs. Time: 6.27090 s. Total ***** plots
2021-06-18T03:58:47.319 harvester chia.harvester.harvester: INFO     ** plots were eligible for farming 56b44ec77f... Found 0 proofs. Time: 4.15971 s. Total ***** plots
2021-06-18T03:58:47.730 harvester chia.harvester.harvester: INFO     ** plots were eligible for farming 56b44ec77f... Found 0 proofs. Time: 2.67811 s. Total ***** plots
```

emlowe committed:
Try to bump CI to run
